yb-voyager/src/srcdb/common.go

```diff
@@ -82,3 +83,43 @@ func processImportDirectives(fileName string) error {
 	}
 	return nil
 }
+// Strip the schema name from the names qualified by the source schema.
+//
+// ora2pg exports SYNONYM objects as VIEWs. The associated CREATE VIEW DDL statements
+// appear in the EXPORT_DIR/schema/synonyms/synonym.sql.
+//
+// The problem is that the view names are qualified with the Oracle's schema name.
+// Unless, the user is importing the schema in an exactly similarly named schema
+// on the target, the DDL statement will fail to import.
+//
+// The following function goes through all the names from the file and replaces
+// all occurrences of `sourceSchemaName.objectName` with just `objectName`.
+func stmtsToStripSourceSchemaNames(fileName string, sourceSchema string) error {
+	if !utils.FileOrFolderExists(fileName) {
+		return nil
+	}
+	// Create a temporary file after appending .tmp extension to the fileName.
+	tmpFileName := fileName + ".tmp"
+	tmpFile, err := os.Create(tmpFileName)
+	if err != nil {
+		return fmt.Errorf("create %q: %w", tmpFileName, err)
+	}
+	defer tmpFile.Close()
```

Comment:
No need to create the file upfront. `os.WriteFile()` will create the file if it doesn't exist:
<img width="569" alt="image" src="https://user-images.githubusercontent.com/98899964/209327563-0dd1cd1e-6130-4563-b3fd-8db6ac806a3d.png">